src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -71,9 +71,12 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
 
     DockerBuildImage() {
         ext.getImageId = { imageId }
-
+        onlyIf {
+            boolean runTask = previouslyBuiltImageExists() ? false : true
+            if (!runTask) { logger.quiet("UP-TO-DATE") }
+            runTask
+        }
```

Comment:
So the 'outputs.upToDateWhen { ... } does not work as expected for custom tasks. Searching around on the internets reveals a few reasons why. Others on the gradle forums suggest the above to get around this limitation. It feels hackish to me but really I don't see any way around it. @bmuschko thoughts?